### PR TITLE
maven: update to 3.5.3

### PR DESCRIPTION
**Pull request by olbohlen**
update to recent maven

### CODE REVIEW

File: components/developer/maven/maven.p5m [hunk 1]
```diff
@@ -19,47 +20,43 @@ set name=info.upstream-url value=$(COMPONENT_PROJECT_URL)
...
 
 file path=usr/share/maven/LICENSE
 file path=usr/share/maven/NOTICE
 file path=usr/share/maven/README.txt
 file path=usr/share/maven/bin/m2.conf
-file path=usr/share/maven/bin/mvn mode=0555
-file path=usr/share/maven/bin/mvnDebug mode=0555
-file path=usr/share/maven/bin/mvnyjp mode=0555
-file path=usr/share/maven/boot/plexus-classworlds-2.5.1.jar
+file path=usr/share/maven/bin/mvn
```
Comment (pyhalov) [inline components/developer/maven/maven.p5m:29]: Check resulting file mode

File: components/developer/maven/maven.p5m [hunk 2]
```diff
@@ -19,47 +20,43 @@ set name=info.upstream-url value=$(COMPONENT_PROJECT_URL)
 set name=info.source-url value=$(COMPONENT_ARCHIVE_URL)
 set name=org.opensolaris.consolidation value=$(CONSOLIDATION)
 
-license LICENSE license='Apache 2.0'
-
-link path=usr/bin/mvn target=../share/maven/bin/mvn
```
Comment (pyhalov) [inline components/developer/maven/maven.p5m:24]: Preserve usr/bin/mvn link

Comment (olbohlen) [reply components/developer/maven/maven.p5m:24]: sorry...I always miss something. should make myself a checklist...:/

**Review by AndWac — COMMENTED** [review]
What is the status of this PR?
The old link is still missing.

You also seem to have removed alp's copyright in the maven.p5m file. I'd recommend to re-add it.

File: components/developer/maven/manifests/sample-manifest.p5m [hunk 3]
```diff
@@ -27,40 +27,44 @@ file path=usr/share/maven/NOTICE
...
+file path=usr/share/maven/lib/guice-4.0-no_aop.jar
+file path=usr/share/maven/lib/jansi-1.17.jar
+file path=usr/share/maven/lib/jansi-native/README.txt
+file path=usr/share/maven/lib/jansi-native/freebsd32/libjansi.so
+file path=usr/share/maven/lib/jansi-native/freebsd64/libjansi.so
+file path=usr/share/maven/lib/jansi-native/linux32/libjansi.so
+file path=usr/share/maven/lib/jansi-native/linux64/libjansi.so
+file path=usr/share/maven/lib/jansi-native/osx/libjansi.jnilib
+file path=usr/share/maven/lib/jansi-native/windows32/jansi.dll
+file path=usr/share/maven/lib/jansi-native/windows64/jansi.dll
```
Comment (AndWac) [inline components/developer/maven/manifests/sample-manifest.p5m:58]: All the jansi-native files must be removed; maybe we need to provide our own for oi?

**Pull request closed without merging**